driver-async/src/main/com/mongodb/async/client/AbstractSubscription.java

```diff
@@ -134,9 +140,12 @@ void onError(final Throwable t) {
         }
     }
 
-    void onNext(final TResult next) {
+    private void onNext(final TResult next) {
         if (!isTerminated()) {
             try {
+                if (next == null) {
+                    throw new UnsupportedOperationException("A null value is invalid");
```

Comment:
I think this should be a MongoClientException, and with a clearer error message as to why null is invalid.

Reply:
Done